tests/epyccel/test_epyccel_division.py

```diff
@@ -145,6 +145,23 @@ def fdiv_i_i(x, y):
     assert (f(-x, y) == fdiv_i_i(-x, y))
     assert (f(x, -y) == fdiv_i_i(x, -y))
     assert (f(-x, -y) == fdiv_i_i(-x, -y))
+    assert isinstance(f(x, y), type(fdiv_i_i(x, y)))
+
+def test_call_fdiv_i_i_i(language):
+    @types(int, int, int)
+    def fdiv_i_i_i(x, y, z):
+        return x // y // z
+
+    f = epyccel(fdiv_i_i_i, language=language)
+    x = randint(1e9)
+    y = randint(low=1, high= 1e3)
+    z = randint(low=1, high= 1e2)
+
+    assert (f(x, y, z) == fdiv_i_i_i(x, y, z))
+    assert (f(-x, y, z) == fdiv_i_i_i(-x, y, z))
+    assert (f(x, -y, z) == fdiv_i_i_i(x, -y, z))
+    assert (f(-x, -y, z) == fdiv_i_i_i(-x, -y, z))
+    assert isinstance(f(x, y, z), type(fdiv_i_i_i(x, y, z)))
```

Comment:
I'm not sure this will be sufficient to test that we have the correct type as the returned type will always be whatever you declared it to be. Maybe there is a compiler warning that you could use to test this? E.g. `epyccel(fdiv_i_i_i, language=language, fflags='-Werror -Wconversion'`
See https://github.com/pyccel/pyccel/blob/master/tests/epyccel/test_kind.py#L34-L45

Reply:
`Wconversion` has been changed in the newer version of gcc ().By using the  flag, we will be obliged to cast all arguments to `double `, because `Wconversion`  checks also for mismatching of the arguments passed to function with the acceoted.

Reply:
the [`Wconversion` ](https://gcc.gnu.org/wiki/NewWconversion)has been changed in the newer version of gcc (4.3 >=).
By using this flag, we will be obliged to cast all arguments to double, or we will mismatch in the type of arguments passed to the `floor` function.
```c
 double floor(double x)
```
so @EmilyBourne only one of your conversations will be solved.
what do you think?
for me, I see that we don't need this check, because the code will have a lot of casts and will not be more readable.

Reply:
Hmmm, I'm not sure. You're right that the code will need cleaner if we have fewer casts, but I think this is an important bug which potentially causes big problems so maybe it is better to have better tests